Lombiq.Hosting.Tenants.Management/Controllers/ShellSettingsEditorController.cs

```diff
@@ -55,16 +65,16 @@ public async Task<IActionResult> Edit(ShellSettingsEditorViewModel model)
             }
         }
 
-        await _tenantConfigSemaphore.WaitAsync(HttpContext.RequestAborted);
-        try
-        {
-            await _tenantConfigSources.SaveAsync(shellSettings.Name, newSettings);
-        }
-        finally
+        // Try to acquire a lock before using the scope, so that a next process gets the last committed data.
+        var (locker, locked) = await _distributedLock.TryAcquireLockAsync("SHELL_SETTINGS_EDITOR_LOCK", TimeSpan.MaxValue);
```

Comment:
Surely we don't want to potentially wait until the heat death of the universe :) (`TimeSpan.MaxValue` is the default anyway). Rather something along the lines of (this is only locking for the save-reload):

```suggestion
        var (locker, locked) = await _distributedLock.TryAcquireLockAsync("SHELL_SETTINGS_EDITOR_LOCK", TimeSpan..FromSeconds(10));
```